### PR TITLE
Update Go to 1.17

### PR DESCRIPTION
**Pull request by tklauser**
First commit updates the Go version for Cilium build and updates the Docker images.

Second commit changes `go.mod` to `go 1.17` to use the new module graph pruning, see https://golang.org/doc/go1.17#graph-pruning 

Third commit adds `//go:build` lines in addition to existing `//+build` lines, as automatically done by `gofmt` in Go 1.17. Note that also some files generated by `deepcopy-gen` were changed. These changes will be lost on the next update until we update to a version of `deepcopy-gen` which includes https://github.com/kubernetes-sigs/controller-tools/pull/595.

### CODE REVIEW

**Comment by tklauser** [comment]
test-me-please

**Review by aanm — APPROVED** [review]
> How many teams do you want to request review?
> Yes

:smile:

**Review by nebril — APPROVED** [review]

File: bpf/tests/prog_test/prog_test.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: Apache-2.0
 // Copyright 2020-2021 Authors of Cilium
 
+//go:build privileged_tests
```
Comment (pchaigno) [inline bpf/tests/prog_test/prog_test.go:4]: They generate that without a whitespace after `//` :scream: :scream_cat:

**Review by pchaigno — APPROVED** [review]

File: bpf/tests/prog_test/prog_test.go (hunk 1 above)
Comment (tklauser) [reply bpf/tests/prog_test/prog_test.go:4]: Yes, that's the canonical style for `//go:` lines. I think to visually distinguish them from non-directive comments. The `// +build` lines were kind of an outlier in that regard.

Comment (pchaigno) [reply bpf/tests/prog_test/prog_test.go:4]: I see. Thanks for the explanation!

**Review by kkourt — APPROVED** [review]

**Comment by tklauser** [comment]
Received 4 reviews from teams covering the most important aspects of the change (@cilium/build, @cilium/ci-structure and @cilium/vendor). The other groups were requested due to the `gofmt` change (3rd commit) changing build tags across the entire code base. All tests passed, thus merging.

**Pull request merged**